### DISCUSSION TITLE
Package org.apache.struts.taglib.tiles changed to org.apache.struts.tiles.taglib

### DISCUSSION THREAD

**Discussion by bharat-bhansali** [ste-gr/struts1-reloaded#26] [Q&A] [answered]
Hello @ste-gr,

We have noticed that package org.apache.struts.taglib.tiles has been changed to org.apache.struts.tiles.taglib.
We are good to make changes in our code base to comply with new package structure but their are other dependencies (ex : jasper compiler) which is referring to org.apache.struts.taglib.tiles.

Can you please help us understand the thought behind this change and is it possible to publish the version with standard package structure?

Thanks,
Bharat

**Comment by ste-gr** [comment, marked as answer]
@bbhansali1
Hi Bharat,

nice to hear that you make good processing to update your code base. I did some research and found that this renaming was done before my work. I found this old upgrade notes from 1.2 to 1.3 where this situation is mentioned:

* [Notes on upgrading from 1.2 to 1.3](https://web.archive.org/web/20150923041220/http://wiki.apache.org/struts/StrutsUpgradeNotes12to13):
> ## 6. Deployment
> * If you experience NoClassDefFoundErrors in JSP code after deploying a Struts 1.3 webapp using the same context name it had under a prior version, try deleting the generated .java and .class files related to the JSP files. In Tomcat, these are found under the $TOMCAT_HOME/work directory.
> * In particular, the package name of the Tiles classes changed from **org.apache.struts.taglib.tiles** to **org.apache.struts.tiles.taglib**, so you may see: java.lang.NoClassDefFoundError: org.apache.struts.taglib.tiles.InsertTag


I'm sorry I couldn't give you a better answer. As I know, other people (myself included) haven't had any problems with JSPs. Maybe I can help you?

Greetings
Stefan

**Reply by bharat-bhansali** [reply]
Thanks @ste-gr, 

This info helps.
